src/msw/joystick.cpp

```diff
@@ -329,16 +329,23 @@ wxString wxJoystick::GetProductName() const
     if (joyGetDevCaps(m_joystick, &joyCaps, sizeof(joyCaps)) != JOYERR_NOERROR)
         return wxEmptyString;
 
+
     wxRegKey key1(wxString::Format(wxT("HKEY_LOCAL_MACHINE\\%s\\%s\\%s"),
                    REGSTR_PATH_JOYCONFIG, joyCaps.szRegKey, REGSTR_KEY_JOYCURR));
 
-    key1.QueryValue(wxString::Format(wxT("Joystick%d%s"),
-                                     m_joystick + 1, REGSTR_VAL_JOYOEMNAME),
-                    str);
+    if (key1.Exists()) {
```

Comment:
Please use wxWidgets coding style, notably put the opening brace on its own line.

Spacing is also typically different in wx code, i.e. this would normally look like

    if ( key1.Exists() )
    {
    ...
    }

This is pretty minor, of course, but style consistency matters in large code bases.

Reply:
Fixed